import { jest, describe, it, expect, beforeEach } from '@jest/globals';
import { CreditcardUtilityService } from '../../src/services/creditcardUtilityService';
import { CrossAccountAPICache } from '../../../common/cross-account/cached-apiinformation';
import { ServerError } from '../../util/serverError';
import { ErrorCodes } from '../../util/errorCodes';

// Mock lodash
jest.mock('lodash', () => ({
    cloneDeepWith: jest.fn(),
    isObject: jest.fn(),
    toLower: jest.fn(),
    some: jest.fn()
}));

// Mock uuid
jest.mock('uuid', () => ({
    v4: jest.fn()
}));

// Mock logger
jest.mock('../../../common/utils/logger', () => ({
    error: jest.fn(),
    trace: jest.fn(),
    info: jest.fn(),
    logBackendServiceRequest: jest.fn(),
    logBackendServiceResponse: jest.fn(),
    logBackendServiceError: jest.fn()
}));



/** @type {import('ts-jest').JestConfigWithTsJest} */
export default {
  preset: "ts-jest",
  testEnvironment: "node",
  // Ignore resource files
  testPathIgnorePatterns: [
    "/node_modules/",
    "/test/resources/"
  ],
  // Specify where test files are located
  testMatch: [
    "**/__tests__/**/*.test.ts",
    "**/?(*.)+(spec|test).ts"
  ],
  extensionsToTreatAsEsm: ['.ts'],
  transform: {
    "^.+\\.tsx?$": [
      "ts-jest",
      {
        useESM: true,
        isolatedModules: true,
      },
    ],
  },
};


-----------------------------------------------------------------

The system was attempting to access account details via the API endpoint:
/channels-ms-api-accinfo/account/v1/accountsdetails
This API call involved:

First making a call to retrieve debit card information from https://api2-sit2.bmogc.net/api/customer-access-entitlement/debit-card/full-report/get
Then making a call to retrieve credit card information from https://api2-sit2.bmogc.net/api/credit-charge-card/credit-card-arrangement-report/get


Looking at the credit card report API call, the system was trying to retrieve data for the following segments: IAS, ICI, ICL, ICY, IGA, IGB, IRA, IRW, ITA, and ITD for card number ending in 2485 (full number: 7719C5165660000192485)
The TSYS credit card system responded with multiple "Unknown Account" errors for this card number across all requested segments, with error code GBL-906.
When the ObjectMapperService.processCCTsysAccounts method tried to process these credit card account responses at line 535, it encountered a NullPointerException because it was not handling the "Unknown Account" error condition properly.
